### PR TITLE
docs: Rewrite README

### PR DESCRIPTION
**Pull request by luise**
This is the first attempt on the new readme. Not looking to merge right now, just looking for feedback for the next iteration. 

Given no major changes, the following will have to be done before we can merge:

- [x] Modify the README to work with the readme_test (we will need to add some "invisible" code snippets -- e.g. import quilt --  to make the example code compile)
- [x] Alias `quilt ps` to `quilt show`
- [x] Check if we are allowed to use cloud provider logos (@ejj)
- [x] Modify the Redis blueprint to export the constructor directly
- [ ] Set up an examples repository with all the composed blueprints (e.g. MEAN and Django/Mongo) -- this can wait, if need be, and the README can just link to a specific blueprint repository instead.
- [x] Make a page for CLI docs (or at least a "Coming soon" page)

### CODE REVIEW

**Comment by luise** [comment]
@ejj or @kklin , I updated this and all the prereqs -- it's ready for review

File: scripts/blueprints-tester/tests/test_readme.go [hunk 1]
```diff
@@ -134,3 +141,12 @@ func TestReadme() error {
 	}
 	return nil
 }
+
+func getMatch(pattern, line string) string {
```
Comment (kklin) [inline scripts/blueprints-tester/tests/test_readme.go]: The more go-ish way to represent "no match" is to return a bool as a second return value.

`blockIDMatch, isBlockID := getMatch(blockIDPattern, line)`

File: README.md [hunk 2]
```diff
@@ -4,138 +4,93 @@
...
-The Quilt.js JavaScript framework allows for development, versioning, and
-testing of infrastructure in the same way we do for application code.
-Additionally, Quilt.js code is shareable, reusable and composable, making it
-easy to set up and manage systems without being an expert in system
-administration.
+## Deploy Quickly on...
 
-Quilt is a research project out of UC Berkeley. It is currently under heavy
-development, but please try it out - we are eager for feedback!
+![providers](./docs/source/images/providers.png)
```
Comment (kklin) [inline README.md]: Looks like this image isn't included?

File: README.md [hunk 3]
```diff
@@ -4,138 +4,93 @@
 
 # Quilt
 
-<img src="https://github.com/quilt/mean/blob/master/images/mean.gif">
+Deploying applications to the cloud can be painful. Booting virtual machines, configuring networks, and setting up databases, requires massive amounts of specialized knowledge — knowledge that’s scattered across documentation, blog posts, tutorials, and source code.
```
Comment (kklin) [inline README.md]: These lines should be word wrapped.

File: README.md [hunk 4]
```diff
@@ -4,138 +4,93 @@
...
-var app = new Node({
-  nWorker: 3,
-  repo: "https://github.com/tejasmanohar/node-todo.git",
-  env: {
-    PORT: "80",
-    MONGO_URI: mongo.uri("mean-example")
-  }
-});
-var haproxy = new HaProxy(3, app.services());
+<!-- const {Container, Service, Machine, publicInternet} = require('@quilt/quilt'); -->
```
Comment (kklin) [inline README.md]: Any reason not to have this closer to the code block?

File: scripts/blueprints-tester/tests/test_readme.go (hunk 1 above)
Comment (ejj) [inline scripts/blueprints-tester/tests/test_readme.go]: 👍

File: README.md (hunk 3 above)
Comment (ejj) [inline README.md]: Btw, in vim, if you highlight the text and type `gq` it will do it for you.

**Comment by ejj** [comment]
LGTM don't have many comments beyond what @kklin already mentioned

File: README.md (hunk 2 above)
Comment (luise) [inline README.md]: Ah, good catch!

File: README.md (hunk 4 above)
Comment (luise) [inline README.md]: To me it kind of signaled more clearly that the import is related to all the code snippets -- but I don't feel strongly about it. I'll move it closer to the code

File: scripts/blueprints-tester/tests/test_readme.go (hunk 1 above)
Comment (luise) [inline scripts/blueprints-tester/tests/test_readme.go]: Oh, that would be much nicer. I'll do that!

**Comment by luise** [comment]
@kklin and @ejj  addressed your comments. Thanks!

File: README.md (hunk 4 above)
Comment (kklin) [inline README.md]: Oh, that makes sense. You can move it back if you like 😄

Comment (luise) [inline README.md]: Haha, nah, it's fine either way!

**Pull request merged**